lib/supavisor/handlers/proxy/handler.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Supavisor.Handlers.Proxy.Handler do
+  @moduledoc false
+
+  require Logger
+
+  @behaviour :ranch_protocol
+  @behaviour :gen_statem
+
+  alias Supavisor.{
+    Helpers,
+    Protocol.Server,
+    Monitoring.PromEx,
+    Handlers.Proxy.Db,
+    Handlers.Proxy.Client
+  }
+
+  alias Supavisor.HandlerHelpers, as: HH
+
+  @sock_closed [:tcp_closed, :ssl_closed]
+  @proto [:tcp, :ssl]
+
+  @impl true
+  def start_link(ref, transport, opts) do
+    pid = :proc_lib.spawn_link(__MODULE__, :init, [ref, transport, opts])
+    {:ok, pid}
```

Comment:
I would use `:proc_lib.start_link/3` instead.

Reply:
Any reason for that? In the cowboy codebase, they mostly use spawn_link

Reply:
Somehow `start_link/3` is cleaner for me, as I can return error from spawned process (and also inform the parent when the handler is ready).